### PR TITLE
[Tech] Util restructure

### PR DESCRIPTION
**Pull request by Nocccer**
This PR restructure `util.ts` and split it into multiple files.
This helps to stop growing of the file and make the code more readable and testable.

- All functions that are related to wine went to a new `utils.ts` file in the `backend/wine` folder
- All functions that are related to gog went to a new `utils.ts` file in the `backend/gog` folder
- All functions that are related to legendary went to a new `utils.ts` file in the `backend/legendary` folder
- The left over functions were splitted into:
  - abort
  - app
  - connection
  - error
  - filesystem
  - format
  - game
  - process
  - runtime
  - system
- To still import function via `import {xyz} from 'backend/utils'` a index.ts was added which export all files under utils folder.

If there is any function or subsection that you think does not fit let me know.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: .github/workflows/codecheck.yml [hunk 1]
```diff
@@ -17,3 +17,5 @@ jobs:
         run: yarn
       - name: Check Typescript syntax
         run: yarn codecheck
+      - name: Find dead code
```
Comment (Nocccer) [inline .github/workflows/codecheck.yml:20]: Remove

File: src/backend/config.ts [hunk 2]
```diff
@@ -29,10 +29,10 @@ import {
   getSteamCompatFolder,
   configStore
 } from './constants'
-import { execAsync } from './utils'
 import { execSync } from 'child_process'
 import { logError, logInfo, LogPrefix } from './logger/logger'
 import { dirname, join } from 'path'
+import { execAsync } from './utils/process/process'
```
Comment (arielj) [inline src/backend/config.ts:35]: if we create an `index.ts` inside the `process` (and the others) and export things we can import things without this duplication, like `import { execAsync } from './utils/process'`

even more, we could also have an index.ts inside `/utils` that exports everything from every file inside it so we can still do `import { execAsync } from './utils'` like it was before and keep the separation in files for testability/maintainability

Comment (Nocccer) [reply src/backend/config.ts:35]: Sounds good to me. Will work on this.

**Comment by Nocccer** [comment]
Closing this. To much conflicts. Probably redo would make more sense.

**Pull request closed without merging**